[section#thread.req.exception
    [:en] Exceptions
]

[para]
[:en] Some functions described in this Clause are specified to throw exceptions
of type [`system_error] ([#syserr.syserr]).

[:en] Such exceptions shall be thrown if any of the function[=']s error
conditions is detected or a call to an operating system or other underlying API
results in an error that prevents the library function from meeting its
specifications.

[:en] Failure to allocate storage shall be reported as described in
[#res.on.exception.handling].

[para:~]
[enter:example]
[:en] Consider a function in this clause that is specified to throw exceptions
of type [`system_error] and specifies error conditions that include
[`operation_not_permitted] for a thread that does not have the privilege to
perform the operation.

[:en] Assume that, during the execution of this function, an [`errno] of
[`EPERM] is reported by a POSIX API call used by the implementation.

[:en] Since POSIX specifies an [`errno] of [`EPERM] when ["the caller does not
have the privilege to perform the operation], the implementation maps [`EPERM]
to an [`error_condition] of [`operation_not_permitted] ([#syserr]) and an
exception of type [`system_error] is thrown.
[exit:example]

[para]
[:en] The [`error_code] reported by such an exception[=']s [`code()] member
function shall compare equal to one of the conditions specified in the
function[=']s error condition element.
